### PR TITLE
Change FRAMEWORK_SEARCH_PATH for xcframeworks

### PR DESCRIPTION
**Pull request by FranzBusch**
We ran into problems when switching from a regular `.framework` to an `.xcframework`. We had a static library target which depended on the `.framework` and after exchanging it for an `.xcframework` we got a compilation error that said `No such module XXX`. 

After investigating a bit, I figured out that due to the `FRAMEWORK_SEARCH_PATH` build setting that is set by XcodeGen for any framework dependency Xcode was not able to locate the proper file anymore. I think this is due to the setup of `.xcframework`s. They include multiple regular `.framework`s inside nested folders. Therefore, we need to pass a recursive path to `FRAMEWORK_SEARCH_PATH` so that it can find all the possible frameworks within the xcframework.

I adapted the code to handle this case, but I am not really sure if this is the proper approach. Would it make sense to create a new dependency type called `.xcframework` instead of special casing the `.framework` type?

It would be great to get your feedback @yonaskolb on this before I start writing tests.

### CODE REVIEW

**Comment by yonaskolb** [comment]
Hi @FranzBusch, thanks for this! Yeah I'm happy with your approach, if you could add some tests and a changelog entry

**Comment by FranzBusch** [comment]
@yonaskolb  I added some tests. Let me know if anything else is needed.

**Comment by yonaskolb** [comment]
Thanks @FranzBusch. Could you add a changelog entry as well please. You'll need to rebase off master, as there has been a release since then

**Comment by FranzBusch** [comment]
@yonaskolb done :)

File: Tests/XcodeGenKitTests/ProjectGeneratorTests.swift [hunk 1]
```diff
@@ -1369,6 +1369,45 @@ class ProjectGeneratorTests: XCTestCase {
                 // generated plist should not be in buildsettings
                 try expect(targetConfig.buildSettings["INFOPLIST_FILE"] as? String) == predefinedPlistPath
             }
+            
+            describe("XCFramework dependencies") {
+                $0.context("with xcframework dependency") {
+                    $0.it("should add FRAMEWORK_SEARCH_PATHS_SEARCH_PATHS") {
```
Comment (yonaskolb) [inline Tests/XcodeGenKitTests/ProjectGeneratorTests.swift:1375]:
```suggestion
                    $0.it("should add FRAMEWORK_SEARCH_PATHS") {
```

File: Tests/XcodeGenKitTests/ProjectGeneratorTests.swift [hunk 2]
```diff
@@ -1369,6 +1369,45 @@ class ProjectGeneratorTests: XCTestCase {
...
+                        let project = Project(name: "test", targets: [app])
+                        let pbxProject = try project.generatePbxProj()
+                        
+                        let target = pbxProject.nativeTargets.first!
+                        let configuration = target.buildConfigurationList!.buildConfigurations.first!
+                        try expect(configuration.buildSettings["FRAMEWORK_SEARCH_PATHS"] as? [String]) == ["$(inherited)", "\"some/folder/MyXCFramework.xcframework/**\""]
+                    }
+                }
+                $0.context("with regular framework") {
+                    $0.it("should add FRAMEWORK_SEARCH_PATHS_SEARCH_PATHS") {
```
Comment (yonaskolb) [inline Tests/XcodeGenKitTests/ProjectGeneratorTests.swift:1393]:
```suggestion
                    $0.it("should add FRAMEWORK_SEARCH_PATHS") {
```

**Pull request merged**